### DISCUSSION TITLE
Ray3f property

### DISCUSSION THREAD

**Discussion by asprasan** [mitsuba-renderer/mitsuba3#481] [Q&A] [answered]
The official documentation states that if I define a `ray = mi.Ray3f(o = origin, d= direction)` then `ray.d` returns `origin`.

Link [here](https://mitsuba.readthedocs.io/en/stable/src/api_reference.html#mitsuba.Ray3f.d)

But, when I use it, `ray.d` actually returns `direction`. I think I just wanted to mention this. I think another property `ray.o` needs to be added to the documentation that actually returns the origin.

**Comment by njroussel** [comment, marked as answer]
Thank you for reporting this @asprasan. This is just a bug with how our documentation is generated.

In your code, you should indeed have `ray.o == origin` and `ray.d == direction`.

**Reply by njroussel** [reply]
This has now been fixed.
